src/topology/local_homeomorph.lean

```diff
@@ -411,52 +430,141 @@ def prod (e : local_homeomorph α β) (e' : local_homeomorph γ δ) : local_home
+    apply continuous_within_at.comp (e.continuous_to_fun (f x) hx) f_cont (inter_subset_right _ _) },
   { assume fe_cont,
-    have A : e.inv_fun '' s ⊆ e.source ∩ e.to_fun ⁻¹' s,
-    { rw [image_subset_iff, preimage_inter],
-      assume x hx,
-      simp [h hx, hx, e.map_target] },
-    have : continuous_on e.inv_fun s := e.continuous_inv_fun.mono h,
-    have : continuous_on ((f ∘ e.to_fun) ∘ e.inv_fun) s := continuous_on.comp fe_cont this A,
-    refine continuous_on.congr_mono this (λx hx, _) (subset.refl _),
-    simp [h hx, hx] }
+    have : continuous_within_at (e.inv_fun ∘ (e.to_fun ∘ f)) (s ∩ f ⁻¹' e.source) x,
+    { have : continuous_within_at e.inv_fun univ (e.to_fun (f x))
+        := (e.continuous_at_inv_fun (e.map_source hx)).continuous_within_at,
+      exact continuous_within_at.comp this fe_cont (subset_univ _) },
+    exact this.congr (λy hy, by simp [e.left_inv hy.2]) (by simp [e.left_inv hx]) }
+end
+
+/-- Continuity at a point can be read under right composition with a local homeomorphism if a
+neighborhood of the initial point is sent to the source of the local homeomorphism-/
+lemma continuous_at_iff_continuous_at_comp_left
+  {f : γ → α} {x : γ} (h : f ⁻¹' e.source ∈ nhds x) :
+  continuous_at f x ↔ continuous_at (e.to_fun ∘ f) x :=
+begin
+  have hx : f x ∈ e.source := (mem_of_nhds h : _),
+  have h' : f ⁻¹' e.source ∈ nhds_within x univ, by rwa nhds_within_univ,
+  rw [← continuous_within_at_univ, ← continuous_within_at_univ,
+      e.continuous_within_at_iff_continuous_within_at_comp_left hx h']
 end
 
 /-- A function is continuous on a set if and only if its composition with a local homeomorphism
 on the left is continuous on the corresponding set. -/
-lemma continuous_on_iff_continuous_on_comp_left {f : γ → α} {s : set γ} (h : f '' s ⊆ e.source) :
+lemma continuous_on_iff_continuous_on_comp_left {f : γ → α} {s : set γ} (h : s ⊆ f ⁻¹' e.source) :
   continuous_on f s ↔ continuous_on (e.to_fun ∘ f) s :=
 begin
   split,
   { assume f_cont,
     exact continuous_on.comp e.continuous_to_fun f_cont h },
   { assume fe_cont,
-    have : e.to_fun ∘ f '' s ⊆ e.target, from calc
-      e.to_fun ∘ f '' s = e.to_fun '' (f '' s) : by rw image_comp
-      ... ⊆ e.to_fun '' (e.source) : image_subset _ h
-      ... = e.target : e.to_local_equiv.image_source_eq_target,
-    have : continuous_on (e.inv_fun ∘ e.to_fun ∘ f) s :=
-      continuous_on.comp e.continuous_inv_fun fe_cont this,
+    have : continuous_on (e.inv_fun ∘ e.to_fun ∘ f) s,
+    { apply continuous_on.comp e.continuous_inv_fun fe_cont,
+      assume x hx,
+      have : f x ∈ e.source := h hx,
+      simp [this] },
     refine continuous_on.congr_mono this (λx hx, _) (subset.refl _),
-    have : f x ∈ e.source := h (mem_image_of_mem _ hx),
+    have : f x ∈ e.source := h hx,
     simp [this] }
 end
 
-end continuous_on
+end continuity
+
+/-- If a local homeomorphism has source and target equal to univ, then it induces a homeomorphism. -/
```

Comment:
This docstring reads as if you were stating a lemma rather than defining an object.

Reply:
Clarified